src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -6005,6 +6005,7 @@ private void VisitConditionalAccess(BoundConditionalAccess node, out PossiblyCon
 
             // Per LDM 2019-02-13 decision, the result of a conditional access "may be null" even if
             // both the receiver and right-hand-side are believed not to be null.
+            UnsplitIfNeeded(resultType);
```

Comment:
>UnsplitIfNeeded(resultType); [](http://example.com/codeflow?start=12&length=28)

Do we understand why we are left in a conditional state? Could you explain why do you think this is the right place for the fix?

Reply:
> Do we understand why we are left in a conditional state?

We analyze the call `""?.M1(x)` and since M1 has post-condition on the argument, it leaves us in conditional state.

> Could you explain why do you think this is the right place for the fix?

VisitConditionalAccess needs to set a result type to maybe-null. The assert in SetResultType says we can call it in conditional state only for `bool`s (which makes sense, other types cannot be conditional). So calling `UnsplitIfNeeded` beforehand ensures we are not in conditional state when the type is not `bool`.